pyccel/codegen/printing/ccode.py

```diff
@@ -542,8 +547,21 @@ def stored_in_c_pointer(self, a):
     def _print_FunctionCall(self, expr):
         func = expr.funcdef
          # Ensure the correct syntax is used for pointers
-        args = [VariableAddress(a) if isinstance(a, Variable) and self.stored_in_c_pointer(f) else a
-            for a, f in zip(expr.arguments, func.arguments)]
+
+        args = []
+        for a, f in zip(expr.arguments, func.arguments):
+            if isinstance(a, Variable) and self.stored_in_c_pointer(f):
+                args.append(VariableAddress(a))
+            elif f.is_optional and not isinstance(a, Nil):
+                tmp_var_name = self._parser.get_new_name('tmp')
+                tmp_var = Variable(name = tmp_var_name, dtype = f.dtype)
```

Comment:
I recommend using `f.clone` if this is possible, that way we are sure that `Variable` includes all the necessary information (e.g. shape, rank, is_pointer, etc)

Reply:
I don't think we need to copy all informations

Reply:
I think this is a problem with the `clone` function, I'm not sure `is_optional` should be copied. However the additional information that is copied is not yet implemented in C, so maybe this is good enough for now.